src/celo/types.ts

```diff
@@ -128,35 +138,62 @@ export type RpcTransactionCIP64<TPending extends boolean = boolean> = Omit<
 > &
   FeeValuesEIP1559<Quantity> & {
     feeCurrency: Address | null
+    maxFeeInFeeCurrency?: undefined
     gatewayFee?: undefined
     gatewayFeeRecipient?: undefined
     type: '0x7b'
   }
 
+export type RpcTransactionCIP66<TPending extends boolean = boolean> = Omit<
+  TransactionBase<Quantity, Index, TPending>,
+  'typeHex'
+> &
+  FeeValuesEIP1559<Quantity> & {
+    feeCurrency: Address | null
+    maxFeeInFeeCurrency: Quantity | null
+    gatewayFee?: undefined
+    gatewayFeeRecipient?: undefined
+    type: '0x7a'
+  }
+
 export type RpcTransactionRequestCIP64 = TransactionRequestBase<
   Quantity,
   Index
 > &
   ExactPartial<FeeValuesEIP1559<Quantity>> & {
     accessList?: AccessList | undefined
     feeCurrency?: Address | undefined
+    maxFeeInFeeCurrency?: undefined
     type?: '0x7b' | undefined
   }
 
+export type RpcTransactionRequestCIP66 = TransactionRequestBase<
+  Quantity,
+  Index
+> &
+  ExactPartial<FeeValuesEIP1559<Quantity>> & {
+    accessList?: AccessList | undefined
+    feeCurrency?: Address | undefined
+    maxFeeInFeeCurrency?: Quantity | undefined
+    type?: '0x7a' | undefined
+  }
+
 type Transaction<TPending extends boolean = boolean> = Transaction_<
   bigint,
   number,
   TPending
 > & {
   feeCurrency: Address | null
-  gatewayFee?: undefined
-  gatewayFeeRecipient?: undefined
+  maxFeeInFeeCurrency: bigint | null
+  gatewayFee: bigint | null
+  gatewayFeeRecipient: Address | null
```

Comment:
this looks wrong. there is no case where both these exist on the same transaction object